changelog.d/pr-7340.md

```diff
@@ -0,0 +1,15 @@
+### 🔩 Dependencies
+
+- Deprecated `boto` is replaced with `boto3` (used to handle AWS S3
+  downloads). Fixes [#5597](https://github.com/datalad/datalad/issues/5597)
+  via [PR #7340](https://github.com/datalad/datalad/pull/7340)
+  (by [@mslw](https://github.com/mslw), [@effigies](https://github.com/effigies), and [@yarikoptic](https://github.com/yarikoptic)). 
+  Remaining issues:
+  - no download progress indication,
+  - no "Range" support (for partial downloads).
+
+### 🏠 Internal
+
+- Retry logic for S3 connections is now handed over to Boto3 and its
+  standard mode, removing our custom method.
+  [PR #7340](https://github.com/datalad/datalad/pull/7340)
```

Comment:
@jwodder please remind me if on such a changelog entry we would somehow make our datalad-release action to make it a `minor` and not just `patch` release?

Reply:
The release action ignores the contents of changelog entries; it only cares about the semver labels.